### PR TITLE
Add minor fixes encountered during testing

### PR DESCRIPTION
**Pull request by VinayLondhe14**
Encountered a few things while testing that need fixing

### CODE REVIEW

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDFactory.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ public RDD<InternalRow> buildScanFromSQL(String sql) {
 
     log.info(
         "Querying table {}, requiredColumns=[{}]",
-        actualTable.getFriendlyName(),
+        actualTable.getTableId().getTable(),
```
Comment (VinayLondhe14) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDFactory.java:98]: `actualTable.getFriendlyName()` return null. `actualTable.getTableId().getTable()` returns the correct name

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala [hunk 2]
```diff
@@ -51,9 +51,6 @@ class BigQueryStrategy(expressionConverter: SparkExpressionConverter, expression
     try {
       generateSparkPlanFromLogicalPlan(plan)
     } catch {
-      case ue: BigQueryPushdownUnsupportedException =>
```
Comment (VinayLondhe14) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala:54]: We do not want to throw an exception here because even though pushdown is unsupported for such operations (such as `CreateViewCommand`), they will be completed in Spark. An exception terminates the entire process and we don't want that to happen.

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 3]
```diff
@@ -295,7 +307,9 @@ Iterable<Table> listTables(DatasetId datasetId, TableDefinition.Type... types) {
   TableId createDestinationTable(
       Optional<String> referenceProject, Optional<String> referenceDataset) {
     String project = materializationProject.orElse(referenceProject.orElse(null));
-    String dataset = materializationDataset.orElse(referenceDataset.orElse(null));
+    String dataset =
```
Comment (VinayLondhe14) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:310]: Open question here. If customers do not pass in a dataset while pushing down a query, do we create one for them?

Here, I have added code that creates a dataset if none is given

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala (hunk 2 above)
Comment (davidrabinowitz) [reply spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala:54]: This should be in a comment. Also, if why don't we want to fall spark it we cannot generate a spark plan due to an exception ? Create view should not be pushed down

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]
I'm not sure the connector needs to create datasets

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java (hunk 3 above)
Comment (VinayLondhe14) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java]: Synced with David. We do not want to create the dataset ourselves for now

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala (hunk 2 above)
Comment (VinayLondhe14) [reply spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala:54]: Added a comment in the code regarding this.

Since this is an all or nothing pushdown, all the commands are tried using the `BigQueryStrategy` first. If BigQueryStrategy returns Nil, it falls back to other Spark strategies. 

We can probably add some optimizations here to fast fail if the passed in plan contains unsupported nodes

Comment (VinayLondhe14) [reply spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala:54]: I have added a method `hasUnsupportedNodes()` that checks for unsupported nodes in the plan. We will need to compare performance with and without the check because with the check we are essentially going through the plan twice.

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDFactory.java (hunk 1 above)
Comment (davidrabinowitz) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDFactory.java:98]: This will print just the table name, not the project and dataset

File: spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala [hunk 4]
```diff
@@ -48,18 +49,35 @@ class BigQueryStrategy(expressionConverter: SparkExpressionConverter, expression
    *         query generation was successful, None if not.
    */
   override def apply(plan: LogicalPlan): Seq[SparkPlan] = {
+    // Check if we have any unsupported nodes in the plan. If we do, we return
+    // Nil and let Spark try other strategies
+    if(hasUnsupportedNodes(plan)) {
```
Comment (davidrabinowitz) [inline spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala:54]: Is `order by` is going to be solved by another PR?

Comment (VinayLondhe14) [reply spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/BigQueryStrategy.scala:54]: yes. this wont solve the `order by` problem. the solution for that will have to be investigated

**Review by davidrabinowitz — APPROVED** [review]

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDFactory.java (hunk 1 above)
Comment (VinayLondhe14) [reply spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/direct/BigQueryRDDFactory.java]: fixed
